packages/calling-stateful-client/review/calling-stateful-client.api.md

```diff
@@ -131,7 +131,7 @@ export interface StatefulCallClient extends CallClient {
 
 // @public
 export type StatefulCallClientArgs = {
-    userId: string;
+    userId?: string;
```

Comment:
Is there any use case where we must have a user id? For Chat we need it to distinguish mine from theirs in messages. In Calling do we want to keep userId to be consistent with Chat?

Reply:
Just asking probing question, I'm fine with making it optional for ease of use for now.

Reply:
We don't have a case where we must have a userId in Calling. This was requested by the team to make the StatefulClient easier to use. We could potentially remove this entirely if we push the responsibility of keeping track of userId to the developer's app. I made this required initially because I felt its not nice dealing with optionals all the time because anytime you use userid you have to first check if it is undefined. I think optionals/undefineds make it harder to reason about the correctness of the code and add more scenarios to consider and be aware of.

Reply:
+1 to avoiding optional wherever the intent is that the field is required.

The opposite is true in APIs -- This value is actually optional. Forcing the user to provide a dummy value is worse than our implementation dealing with optionals.

Re Dominik's question: At least so far we claim that none of the pure components ever use this value. The state merely holds this value for Contoso's (or the Composite's, in practice) use. So likely will stay optional.